qiskit/circuit/library/standard_gates/x.py

```diff
@@ -959,6 +980,9 @@ def inverse(self):
         """Invert this gate. The C4X is its own inverse."""
         return C4XGate(ctrl_state=self.ctrl_state)
 
+    def __eq__(self, other):
+        return isinstance(other, C4XGate) and self.ctrl_state == other.ctrl_state
+
 
 class MCXGate(ControlledGate):
```

Comment:
This might well want to be a follow-up, but if we could get a decent way to improving the `MCXGate` equality checks, it'd help (part of) #10641.